### PR TITLE
Add answer testing framework

### PR DESCRIPTION
**Pull request by brittonsmith**
This adds a framework for performing answer testing. The `EnzoETest` class is a base class for creating answer tests. Test files are run in a temporary directory which is then removed when the tests are done. The `TestHLLCCloud` class gives an example of using this with a simple test problem.

I have added documentation for running and creating new answer tests. This required some slight reorganization and rewording of the existing testing documentation. I've removed instructions from this comment so reviewers can try the instructions in the documentation. Go [here](https://enzo-e--220.org.readthedocs.build/en/220/) to see a rendered version of the documentation. In particular, the new documentation can be found [here](https://enzo-e--220.org.readthedocs.build/en/220/tests/index.html).

This is now working within circleci with a single test. @mabruzzo, I used one of your vlct simulations to provide a template. Please, feel free to alter that test however you see fit. I implemented something very simple.

### CODE REVIEW

**Comment by jwise77** [comment]
Typo in your test filename `test_vclt.py` ➡️ `test_vlct.py`

**Comment by jwise77** [comment]
Comment from @mabruzzo during triage: How and can the tests be run in both single and double precision?  Similar to how we test the compilation in both modes.

**Comment by brittonsmith** [comment]
> Comment from @mabruzzo during triage: How and can the tests be run in both single and double precision? Similar to how we test the compilation in both modes.

I've got a solution for this that works with setting environment variables. I will make sure to include this in the documentation.

**Review by fearmayo — APPROVED** [review]

**Review by mabruzzo — APPROVED** [review]
Sorry for taking so long to review this. This looks great to me!

Two very minor points:
- I did have a question about a website-documentation change that you made.
- I also wanted to ask what you thought about enabling the answer-tests on CI for single precision. In the past, I've anecdotally found that the single precision tests catch a fair amount of mistakes (since I primarily develop the code in double precision mode). It's not strictly necessary to do this right now since we don't have many pytest-tests. But, it might be easier to address this now (so that mistakes in newly implemented answer tests are caught more quickly). What does Enzo do?

File: doc/source/conf.py [hunk 1]
```diff
@@ -268,11 +268,11 @@
...
-html_context = {
-    'css_files': [
-        '_static/theme_overrides.css',  # overrides for wide tables in RTD theme
-        ],
-    }
+# html_context = {
+#     'css_files': [
+#         '_static/theme_overrides.css',  # overrides for wide tables in RTD theme
+#         ],
+#     }
```
[the remark this answers is not in the document]
Comment (brittonsmith) [reply doc/source/conf.py:275]: Ah, sorry! I did this locally in a single commit that I meant to pick out before issuing the PR. I couldn't get the docs to build locally with this in there. I'll pluck the commit and update.

**Comment by brittonsmith** [comment]
@mabruzzo I'm happy to run single precision answer tests on circleci. I wasn't sure we wanted to use resources for both and chose arbitrarily.

**Pull request merged**